### PR TITLE
Merge master into rc

### PR DESCRIPTION
**Pull request by kjcho-msft**

### CODE REVIEW

File: DataProcessing/Spark.nuspec [hunk 1]
```diff
@@ -20,16 +20,17 @@
...
 <file src="**\azure-keyvault-webkey-1.1.jar" target="lib" />
-<file src="**\datax-core_2.3_2.11-1.1.0.jar" target="lib" />
-<file src="**\datax-host_2.3_2.11-1.1.0.jar" target="lib" />
-<file src="**\datax-utility_2.3_2.11-1.1.0.jar" target="lib" />
-<file src="**\datax-keyvault_2.3_2.11-1.1.0-with-dependencies.jar" target="lib" />
-<file src="**\datax-udf-samples_2.3_2.11-1.1.0.jar" target="lib" />
+<file src="**\datax-core*.jar" target="lib" />
+<file src="**\datax-host*.jar" target="lib" />
+<file src="**\datax-utility*.jar" target="lib" />
+<file src="**\datax-keyvault*.jar" target="lib" />
```
Comment (carlbrochu) [inline DataProcessing/Spark.nuspec]: @rohit489 can you look into fixing this before we ship?

File: DataProcessing/datax-utility/pom.xml [hunk 2]
```diff
@@ -50,11 +50,11 @@ SOFTWARE
     </scm>
 
     <groupId>com.microsoft.datax</groupId>
-    <artifactId>datax-utility_2.3_2.11</artifactId>
-    <version>1.1.0</version>
+    <artifactId>datax-utility_2.4_2.11</artifactId>
```
Comment (carlbrochu) [inline DataProcessing/datax-utility/pom.xml]: did we rev our component versions, and are we avoiding snapshots in RC?

Comment (kjcho-msft) [reply DataProcessing/datax-utility/pom.xml]: yes we did. and yes we don't have snapshots in RC.

File: DataProcessing/datax-host/src/main/scala/datax/host/BlobBatchingHost.scala [hunk 3]
```diff
@@ -7,47 +7,65 @@ package datax.host
...
 
     result
   }
 
-  def runBatchApp(inputArguments: Array[String],processorGenerator: UnifiedConfig=>BlobPointerProcessor ) = {
-    val appLog = LogManager.getLogger("runBatchApp")
+  // Get the datetime pattern like {yyyy-MM-dd} in the input path.
+  // For e.g. if the input path is "wasbs://outputs@myaccount.blob.core.windows.net/{yyyy-MM-dd}/flow1", this will return yyyy-MM-dd
+  private def getDateTimePattern(inputPath:String):String ={
+    val regex = """\{([yMdHmsS\-/.]+)\}*""".r
```
Comment (unrealord) [inline DataProcessing/datax-host/src/main/scala/datax/host/BlobBatchingHost.scala]: Can we considering caching this pattern probably as a member of BlobBatchingHost object, or some other common place if this regex could be used for other parts of the code

**Pull request merged**